### ISSUE TITLE
[BUG]iOS: Player crashes when opened / closed multiple times  

### ISSUE THREAD

**Issue by MiroLiebschner** [jhomlala/betterplayer#303]
**Describe the bug**
[iOS real Device only] When BetterPlayerNotificationConfiguration is on and the BetterPlayer is opened / closed multiple times in quick succession the app crashes & closes. The BetterPlayer is not usable after the first crash (app instant crashes again). When the notificationConfiguration is commented out the problems solved. 

Code used: 
```
notificationConfiguration: BetterPlayerNotificationConfiguration(
            showNotification: true,
            title: title,
            author: authorName,
            imageUrl:
                imageUrl),
```

1. Use an environment in which you can close the page the player is on in quick succession.
2. Close and reopen the player a few times in quick succession. 

I would expect the page to just be rebuild every time. Most of the time this is working but sometime the app crashes. 

Error Logs 
Xcode Error Log (please ignore the "timer stopped" line)
<img width="422" alt="Bildschirmfoto 2021-02-15 um 11 26 00" src="https://user-images.githubusercontent.com/56092186/107961981-d1cc1800-6fa6-11eb-8974-77f9b6b72b7a.png">
AndroidStudio 
Just "Lost Connection to Device" in the console. 

Flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 1.22.6, on macOS 11.1 20C69 darwin-x64, locale de-DE)
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 12.4)
[✓] Android Studio (version 3.6)
[✓] Connected device (1 available)

**Better Player version**
- Version: 0.0.52 && 0.0.55

Tested on
 - Device: iPhone SE (2020)
 - OS: 14.2

I was not able to reproduce the error in the example app. I suspect that is because I was not able to open / close the notification player fast enough. In the example app I managed to have multiple versions of the same video running with no ability to control some of the versions (as 1 version is controlled in lockscreen and 1 version is controlled via the last opened BetterPlayerPage but 3. and 4. versions will just keep playing).

**Comment by jhomlala** [comment]
Please provide reproducible example and steps which need to be done. Please make this as simplest as you can. Probably there's some issue but I need to debug it on my side.

**Comment by MiroLiebschner** [comment]
@jhomlala I tried to set up a repository but managed only 1 crash and since then it is working fine (in that minimal repo). So So the error could be on my side. 
But I have better Logs from xCode maybe they are helping to understand the issue: 
Original Exception Backtrace: 
<img width="957" alt="Bildschirmfoto 2021-02-20 um 09 57 56" src="https://user-images.githubusercontent.com/56092186/108581754-1e767100-7362-11eb-94fd-3783c4363e1c.png">
<img width="973" alt="Bildschirmfoto 2021-02-20 um 09 58 38" src="https://user-images.githubusercontent.com/56092186/108581780-364df500-7362-11eb-893d-a2235bc234e4.png">

So I suspect the crash is connected with the imageUrl attribute but only guessing :D

**Comment by jhomlala** [comment]
I've added potential fix for this issue in 0.0.56. Hope it will help. Let me know if you encounter this issue after update.

**Comment by MiroLiebschner** [comment]
Just tested it and the problem seems to be fixed. Thank you very much, great work!
